web/src/routes/(user)/locked/[[photos=photos]]/[[assetId=id]]/+page.ts

```diff
@@ -0,0 +1,37 @@
+import { AppRoute } from '$lib/constants';
+import { authenticate } from '$lib/utils/auth';
+import { getFormatter } from '$lib/utils/i18n';
+import { getAssetInfoFromParam } from '$lib/utils/navigation';
+import { getAuthStatus } from '@immich/sdk';
+import { redirect } from '@sveltejs/kit';
+import type { PageLoad } from './$types';
+
+export const load = (async ({ params, url }) => {
+  await authenticate();
+  const { isElevated, pinCode } = await getAuthStatus();
+  let extractedPath: string | undefined;
+  const regex = /\/locked\/(photos\/[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12})/;
+  const match = url.pathname.match(regex);
+
+  if (match && match[1]) {
+    extractedPath = match[1];
+  }
+
+  if (!isElevated || !pinCode) {
+    let redirectPath = `${AppRoute.AUTH_PIN_PROMPT}`;
+    if (extractedPath) {
+      redirectPath += `?continue=${extractedPath}`;
+    }
```

Comment:
this seems kind of crazy. I think you could just do
```ts
const redirectPath = `${AppRoute.AUTH_PIN_PROMPT}?continue=${url.pathname}`;
```

It's possible it needs to be `encodeURIComponent(url.pathname)`